src/mpid/ch4/netmod/ofi/ofi_noinline.h

```diff
@@ -83,4 +83,27 @@ int MPIDI_OFI_mpi_win_free_hook(MPIR_Win * win);
 #define MPIDI_NM_mpi_win_free_hook MPIDI_OFI_mpi_win_free_hook
 #endif
 
+/* ofi_init.h */
+int MPIDI_OFI_mpi_init_hook(int rank, int size, int appnum, int *tag_bits, MPIR_Comm * comm_world,
+                            MPIR_Comm * comm_self, int spawned, int *n_vcis_provided);
+int MPIDI_OFI_mpi_finalize_hook(void);
+int MPIDI_OFI_get_vci_attr(int vci);
+void *MPIDI_OFI_mpi_alloc_mem(size_t size, MPIR_Info * info_ptr);
+int MPIDI_OFI_mpi_free_mem(void *ptr);
+int MPIDI_OFI_get_local_upids(MPIR_Comm * comm, size_t ** local_upid_size, char **local_upids);
+int MPIDI_OFI_upids_to_lupids(int size, size_t * remote_upid_size, char *remote_upids,
+                              int **remote_lupids);
+int MPIDI_OFI_create_intercomm_from_lpids(MPIR_Comm * newcomm_ptr, int size, const int lpids[]);
+
+#ifdef NETMOD_INLINE
+#define MPIDI_NM_mpi_init_hook MPIDI_OFI_mpi_init_hook
```

Comment:
Maybe a super dumb question -- but what was the reason why we can't compile out the wrapper implementations on `NETMOD_INLINE` builds?

Reply:
The problem is sort of backward from the case when `NETMOD_INLINE` disabled---both `netmod_impl.c/h` and the `ofi_init.h/c` will have their own `MPIDI_NM_*` (the one in `netmod_impl.c/h` is the wrapper). To solve this name collision, we have to change the name of the uninlined functions to `MPIDI_OFI_*`. Then we run into the problem when `NETMOD_INLINE` is enabled, the function name does not match the defined interface name anymore, hence the `#define` solution.

Reply:
Oh I see. I think I remember what we've figured out before. Thanks.